packages/devtools_server/lib/src/client_manager.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:sse/server/sse_handler.dart';
+
+class ClientManager {
+  final List<DevToolsClient> _clients = [];
+  void acceptClient(SseConnection connection) {
+    _clients.add(new DevToolsClient(connection));
+
+    // TODO(dantup): How do we know when a connection goes away to remove it?
+  }
+
+  /// Finds an active DevTools instance that is not already connecting to
+  /// a VM service that we can reuse (for example if a user stopped debugging
+  /// and it disconnected, then started debugging again, we want to reuse
+  /// the open DevTools window).
+  Future<DevToolsClient> findReusableClient() async {
+    // TODO(dantup):
```

Comment:
I would suggest have the clients send messages when a client becomes reusable and when it stops being reusable.
For example, a client could be reusable and then the user could connect to another VM Service or drag in a snapshot of a CPU Profile in which case it is not reusable.

Reply:
Yeah, my plan here was to add a boolean in `DevToolsClient` that tracks whether that client is connected to a VM (and have DevTools ping on connect/disconnect), then have this code just find the first one that's not.

Reply:
> * It required exposing the list of clients in the server API (this might not be a bad thing - maybe there's some future value in editors being able to enumerate the list of open DevTools tabs for the server that it owns - for now the reason is just to help test this is working - we could prefix it with an underscore or something if we'd prefer to consider it "private")

That is fine.